taskfiles/utils-cmake.yaml

```diff
@@ -88,3 +99,66 @@ tasks:
         {{- range .EXTRA_ARGS}}
         "{{.}}"
         {{- end}}
+
+  # Downloads a CMake project tar file from `URL` and then generates, builds, and installs the
+  # project. We purposely omit `sources` and `generates` as we defer to `cmake` to decide whether it
+  # should perform any actions. However, the download and extraction will be skipped if unnecessary.
+  #
+  # @param {string} NAME Project name used for directory names.
+  # @param {string} FILE_SHA256 Content hash to verify the downloaded tar file against.
+  # @param {string} URL
+  # @param {string[]} [BUILD_ARGS] Any additional arguments to pass to the build command.
+  # @param {string} [BUILD_DIR={{.WORK_DIR}}/{{.NAME}}-build] Directory in which to generate the
+  # build system and perform the build.
+  # @param {string[]} [GEN_ARGS] Any additional arguments to pass to the generate command.
+  # @param {string[]} [INSTALL_ARGS] Any additional arguments to pass to the install command.
+  # @param {string} [INSTALL_PREFIX={{.WORK_DIR}}/{{.NAME}}-install] Path prefix of where the
+  # project should be installed.
+  # @param {int} [JOBS] The maximum number of concurrent processes to use when building. If
+  # omitted, the native build tool's default number is used. See `man cmake`.
+  # @param {string} [SOURCE_DIR={{.WORK_DIR}}/{{.NAME}}-src] Directory in which to extract the tar
+  # file.
+  # @param {string[]} [TARGETS] A list of specific targets to build instead of the default target.
+  # @param {string} [WORK_DIR={{.TASK_DIR}}] Base directory to store the build, install, and src
+  # directories inside.
+  cmake-install-remote-tar:
+    internal: true
+    label: "{{.TASK}}:{{.NAME}}-{{.URL}}-{{.INSTALL_PREFIX}}"
+    vars:
+      BUILD_DIR: >-
+        {{default (printf "%s/%s-build" .WORK_DIR .NAME) .BUILD_DIR}}
+      INSTALL_PREFIX: >-
+        {{default (printf "%s/%s-install" .WORK_DIR .NAME) .INSTALL_PREFIX}}
+      SOURCE_DIR: >-
+        {{default (printf "%s/%s-src" .WORK_DIR .NAME) .SOURCE_DIR}}
+      WORK_DIR: >-
+        {{default .TASK_DIR .WORK_DIR}}
+    requires:
+      vars: ["NAME", "FILE_SHA256", "URL", "WORK_DIR"]
```

Comment:
I think `WORK_DIR` is not required since it has a default value based on the task description